hazelcast-spring/src/main/java/com/hazelcast/spring/AbstractHazelcastBeanDefinitionParser.java

```diff
@@ -468,50 +461,45 @@ protected void handleDiscoveryStrategies(Node node, BeanDefinitionBuilder joinCo
-            Assert.isTrue(className != null || implementation != null,
-                    "One of 'class-name' or 'implementation' attributes is required "
-                            + "to create NodeFilter!");
+            NamedNodeMap attributes = node.getAttributes();
+            Node classNameNode = attributes.getNamedItem("class-name");
+            String className = getTextContent(classNameNode).trim();
+            Node implNode = attributes.getNamedItem("implementation");
+            String implementation = getTextContent(implNode).trim();
+            isTrue(!className.isEmpty() || !implementation.isEmpty(),
+                    "One of 'class-name' or 'implementation' attributes is required to create NodeFilter!");
             discoveryConfigBuilder.addPropertyValue("nodeFilterClass", className);
-            if (implementation != null) {
+            if (!implementation.isEmpty()) {
                 discoveryConfigBuilder.addPropertyReference("nodeFilter", implementation);
             }
         }
 
-        private void handleDiscoveryStrategy(Node node, ManagedList discoveryStrategyConfigs) {
-            final BeanDefinitionBuilder discoveryStrategyConfigBuilder =
-                    createBeanBuilder(DiscoveryStrategyConfig.class);
-            final NamedNodeMap attrs = node.getAttributes();
-            Node classNameNode = attrs.getNamedItem("class-name");
-            String className = classNameNode != null ? getTextContent(classNameNode) : null;
-            Node implNode = attrs.getNamedItem("discovery-strategy-factory");
-            String implementation = implNode != null ? getTextContent(implNode) : null;
-            Assert.isTrue(className != null || implementation != null,
-                    "One of 'class-name' or 'implementation' attributes is required "
-                            + "to create DiscoveryStrategyConfig!");
-            if (implementation != null) {
+        private void handleDiscoveryStrategy(Node node, ManagedList<BeanDefinition> discoveryStrategyConfigs) {
+            BeanDefinitionBuilder discoveryStrategyConfigBuilder = createBeanBuilder(DiscoveryStrategyConfig.class);
+            NamedNodeMap attributes = node.getAttributes();
+            Node classNameNode = attributes.getNamedItem("class-name");
+            String className = getTextContent(classNameNode).trim();
+            Node implNode = attributes.getNamedItem("discovery-strategy-factory");
+            String implementation = getTextContent(implNode).trim();
+            isTrue(!className.isEmpty() || !implementation.isEmpty(),
+                    "One of 'class-name' or 'implementation' attributes is required to create DiscoveryStrategyConfig!");
+            if (!implementation.isEmpty()) {
                 discoveryStrategyConfigBuilder.addConstructorArgReference(implementation);
-            } else if (className != null) {
+            } else {
```

Comment:
We use the `getTextContent(node).trim()` pattern in a lot of `handle()` methods already. The `getTextContent()` returns an empty String instead of `null`, so combined with `trim()` we can use `isEmpty()` instead of a manual `null` check.

The change was motivated by a warning from IDEA on this `className != null` check (which is always `true`). I also aligned the other discovery related `handle()` methods.